deploy/dataflow/setup.py

```diff
@@ -0,0 +1,67 @@
+#
+# Copyright (c) Granulate. All rights reserved.
+# Licensed under the AGPL3 License. See LICENSE.md in the project root for license information.
+#
+
+import subprocess
+from distutils.command.build import build as _build  # type: ignore
+
+import setuptools
+
+# ------------------------------------------------------------------------------------------------------------
+# TODO: replace with the actual service name you wish to use
+SERVICE_NAME = "<SERVICE NAME>"
+# TODO: replace with a token you got from the performance studio (https://profiler.granulate.io/installation)
+GPROFILER_TOKEN = "<TOKEN>"
+# ------------------------------------------------------------------------------------------------------------
+
+KEEP_LOGS = True
+
+
+class build(_build):
+    sub_commands = _build.sub_commands + [("ProfilerInstallationCommands", None)]
+
+
+COMMANDS = [
+    "wget -q https://github.com/Granulate/gprofiler/releases/latest/download/gprofiler -O /tmp/gprofiler",
+    "sudo chmod +x /tmp/gprofiler",
+    f"sudo nohup setsid /tmp/gprofiler -cu --token {GPROFILER_TOKEN} --service-name {SERVICE_NAME}"
+    f" --disable-pidns-check > {'/tmp/gprofiler.log' if KEEP_LOGS else '/dev/null'} 2>&1 &",
+]
+
+
+class ProfilerInstallationCommands(setuptools.Command):
+    def initialize_options(self):
+        pass
+
+    def finalize_options(self):
+        pass
+
+    @staticmethod
+    def run_command(command):
+        print("Running command: %s" % command)
+        p = subprocess.Popen(
+            command, stdin=subprocess.PIPE, stdout=subprocess.PIPE, stderr=subprocess.STDOUT, shell=True
+        )
+        stdout_data, _ = p.communicate(input=b'\n')
```

Comment:
You can use subprocess.run instead - https://stackoverflow.com/questions/39187886/what-is-the-difference-between-subprocess-popen-and-subprocess-run/39187984

Reply:
I don't think it has any advantages here. It will work because the `setsid` makes the process run in the background, but I don't think there's a reason to do it. Correct me if I'm wrong though (I'm still trying to avoid doing anything that might break this, and that's a pretty risky change)